src/storage/btree.c

```diff
@@ -25289,7 +25289,9 @@ btree_range_scan (THREAD_ENTRY * thread_p, BTREE_SCAN * bts, BTREE_RANGE_SCAN_PR
 	  if (bts->force_restart_from_root)
 	    {
 	      /* Couldn't advance. Restart from root. */
-	      assert (bts->use_desc_index);
+	      /* TODO: not sure why this assert is present here. However, on passive transaction
+	       * server, it is hit when a page desynchronization situation is encountered */
+	      assert (bts->use_desc_index || is_passive_transaction_server ());
```

Comment:
Before page ahead context, restart from root event occurred only in context of descending index scan.

Reply:
Added a, hopefully, clarifying comment.